src/oscar/views/generic.py

```diff
@@ -27,7 +27,13 @@ def post(self, request, *args, **kwargs):
                 return self.response
             else:
                 messages.error(request, _("Invalid form submission"))
-        return super().post(request, *args, **kwargs)
+        try:
+            f = super().post
+        except AttributeError:
+            messages.error(request, _("Invalid form submission"))
```

Comment:
It's possible to end up adding this message twice now (if `action` exists in the POST data but it's not a valid action, it will be added in the `else` block above, as well as here). That can be resolved by returning `get()` immediately inside that block.

Reply:
Done